src/components/PokemonsList/ListPage/ListPage.js

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from "react";
+import {
+  useParams,
+  useLoaderData,
+  useNavigate,
+  useLocation,
+} from "react-router-dom";
+import { Pagination } from "@mui/material";
+import { FetchingDetailData } from "../FetchingDetailData/FetchingDetailData";
+
+export function ListPage() {
+  const ListLayout = {
+    display: "grid",
+    gridTemplateColumns: "1fr 1fr 1fr 1fr",
+    gap: "40px",
+    marginBottom: "40px",
+    marginTop: "60px",
+  };
+  const PaginationStyle = {
+    marginBottom: "30px",
+  };
+  const { number } = useParams();
+  const Data = useLoaderData();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [pokeName, setPokeName] = useState([]);
+
+  useEffect(() => {
+    setPokeName(Data.results.map((p) => p.name));
+  }, [number]);
+
+  const handleChange = (e, p) => {
+    navigate(`/react-pokedex/pokedex/list/` + p.toString());
+  };
+
+  let currentLink = [];
+  const crumbs = location.pathname.split("/").map((crumb) => {
+    currentLink.push(crumb);
+  });
+  let pageNumber = currentLink[currentLink.length - 1];
+
+  return (
+    <>
+      <div style={ListLayout}>
+        {pokeName.map((name) => (
+          <FetchingDetailData key={name} pokemon={name} />
```

Comment:
Name of that component is confusing. Those are basically Pokemon data, so that should be the name (or something like that). It's ok to handle loading and error state in that called component